core/deployment/src/main/java/io/quarkus/deployment/steps/ClassTransformingBuildStep.java

```diff
@@ -226,9 +231,39 @@ TransformedClassesBuildItem handleClassTransformation(List<BytecodeTransformerBu
                 }
             }
         }
+
+        if (launchMode.getLaunchMode() == LaunchMode.NORMAL) {
+            // the idea here is to write the transformed classes into the build tool's output directory to make core coverage work
```

Comment:
The transformed classes don't end up in the module's jar, right?

Reply:
Which jar are you talking about?

Reply:
They end up in the `transformed-bytecode` jar

Reply:
Ok, that answers the question. Thanks!

Reply:
Does this mean the Jacoco classes can end up in the final production application?

Reply:
The production jar is not affected in any way. This just updates the class files in the build tools output directory.

Reply:
But if you then ran `mvn quarkus:build` a second time without recompiling would this mean Jacoco transformed classes end up in the second build?

Reply:
Ah, no I see what you mean.

I didn't recall that the Jacoco extension itself performs bytecode transformation during the build. But that happens only in test mode, right?
This change is only for the prod run. But basically if you run `mvn quarkus:build` a second time, Quarkus will already have all the application bytecode transformed , which might indeed cause problems for something like Panache.
Do you envision users running this goal standalone?

Reply:
I am also thinking that it might cause problems if you just run `mvn package` instead of `mvn clean package`, if maven/gradle decide not to recompile the class files.

Reply:
Yeah... that could potentially cause issues...

Reply:
@famod I don't support Maven is smart enough to know to recompile when `target/classes` has been updated outside its control, is it?

Reply:
@geoand no it doesn't.

Reply:
Then I think we'll have to ditch this approach unfortunately...